packages/distributor/solana/clients/SolanaAlignedDistributorClient.ts

```diff
@@ -0,0 +1,130 @@
+
+  constructor({ clusterUrl, cluster, commitment, programId, sendRate, sendThrottler }: IInitOptions) {
+    super({ clusterUrl, cluster, commitment, programId, sendRate, sendThrottler });
+    const alignedAirdropsProgram = {
+      ...StreamflowAlignedAirdropsIDL,
+    } as AlignedAirdropsProgramType;
+    this.alignedProxyProgram = new Program(alignedAirdropsProgram, { connection: this.connection });
+  }
+
+  public async getAlignedDistributorData(distributorAddress: string): Promise<AlignedDistributorData | null> {
+    const distributorKey = new PublicKey(distributorAddress);
+    const alignedDistributorKey = getAlignedDistributorPDA(this.alignedProxyProgram.programId, distributorKey);
+    const alignedProxy = await this.alignedProxyProgram.account.alignedDistributor.fetch(alignedDistributorKey);
+    if (!alignedProxy) {
+      throw new Error("Aligned Distributor proxy account not found");
+    }
+
+    const oracleType = Object.keys(alignedProxy.priceOracleType).find((key) => !!key) as OracleTypeName;
+
+    return {
+      oracleType,
+      minPrice: getNumberFromBN(alignedProxy.minPrice, ALIGNED_PRECISION_FACTOR_POW),
+      maxPrice: getNumberFromBN(alignedProxy.maxPrice, ALIGNED_PRECISION_FACTOR_POW),
+      minPercentage: getNumberFromBN(alignedProxy.minPercentage, ALIGNED_PRECISION_FACTOR_POW),
+      maxPercentage: getNumberFromBN(alignedProxy.maxPercentage, ALIGNED_PRECISION_FACTOR_POW),
+      priceOracle: oracleType === "none" ? undefined : alignedProxy.priceOracle.toBase58(),
+      sender: alignedProxy.admin.toBase58(),
+      updatePeriod: getNumberFromBN(alignedProxy.updatePeriod, ALIGNED_PRECISION_FACTOR_POW),
+      clawedBack: alignedProxy.distributorClawedBack,
+    };
+  }
+
+  public async prepareNewDistributorInstruction(
+    data: ICreateAlignedDistributorData,
+    accounts: NewDistributorAccounts,
+  ): Promise<TransactionInstruction> {
+    const { distributor, mint, clawbackReceiver, tokenProgram, tokenVault } = accounts;
+
+    const baseArgs = this.getNewDistributorArgs(data);
+    const alignedArgs = this.getNewAlignedDistributorArgs(data);
+
+    const newDistributorIx = await this.alignedProxyProgram.methods
+      .newDistributor({
+        ...baseArgs,
+        ...alignedArgs,
+      })
+      .accounts({
+        tokenVault,
+        distributor,
+        clawbackReceiver,
+        mint,
+        tokenProgram,
+        priceOracle: new PublicKey(data.oracleAddress),
+      })
+      .instruction();
+
+    return newDistributorIx;
+  }
+
+  public async prepareClawbackInstruction(accounts: ClawbackAccounts): Promise<TransactionInstruction> {
+    const { distributor, from, to, mint, tokenProgram } = accounts;
+    const alignedDistributorKey = getAlignedDistributorPDA(this.alignedProxyProgram.programId, distributor);
+
+    const alignedProxy = await this.alignedProxyProgram.account.alignedDistributor.fetch(alignedDistributorKey);
+    if (!alignedProxy) {
```

Comment:
nit: if you want, you can use `invariant` from "@streamflow/common"
1 line instead of 3

Reply:
oh nice, I will make that change in other places I find as well